Description: fix tempDir no longer defined when EXIT trap is executed
Author: Tom Deseyn <tom.deseyn@gmail.com>
Origin: upstream, https://github.com/dotnet/dotnet/pull/3248
Forwarded: not-needed
Last-Update: 2025-11-12
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/eng/source-build-toolset-init.sh
+++ b/eng/source-build-toolset-init.sh
@@ -71,7 +71,7 @@ function source_only_toolset_init() {
     elif [ -f "$sourceBuiltArchive" ]; then
       # Extract safely into a private temporary directory and ensure cleanup.
       tempDir="$(mktemp -d)"
-      trap 'rm -rf "${tempDir}"' EXIT
+      trap "rm -rf '${tempDir}'" EXIT # tempDir is local, use double quotes to expand.
       tar -xzf "$sourceBuiltArchive" -C "$tempDir" PackageVersions.props
       packageVersionsPath="$tempDir/PackageVersions.props"
     fi
